core/src/main/java/org/apache/accumulo/fate/ZooStore.java

```diff
@@ -529,4 +527,16 @@ public long timeCreated(long tid) {
       return dops;
     }
   }
+
+  /**
+   * Attempt to cancel the reserved transaction by interrupting the Thread that is running it
+   *
+   * @param tid
+   *          transaction id
+   */
+  public void cancel(long tid) {
+    if (reserved.containsKey(tid)) {
```

Comment:
synchronization is needed in this method.

Reply:
Addressed in d5df49354bf83d9174445c5e669936bb5cdf473e